src/io.rs

```diff
@@ -0,0 +1,76 @@
+//! Provides wrappers for Read and Write types which checksum the bytes being read/written.
+use std::io::{Read, Write};
+
+use crate::crc32c_append;
+
+/// Reader wrapper which tracks the checksum of all bytes read.
+pub struct Crc32cReader<R: Read> {
```

Comment:
Bikeshedding here: the technically this doesn't really "read" crc32c values, but rather calculates the checksum as the underlying `Read` implementation is utilized.

I wonder if the right naming here is something like `Crc32cChecksum` or something similar. Thoughts?

Reply:
True, it doesn't read crc32s exactly, but it is a reader, which also takes the Crc32c, which is what I was trying to get at. My use case, at least, was to actually use the bytes they get out while checksumming them. It also reflects nicely with the `Crc32cHasher` too. It may be harder to distinguish between the reader and writer without having those explicitly in the names.

I'm happy to change to whatever you want! I went back and forth with names a *lot* in my checkpipe crate.